spec/services/shift_assignment_service_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe ShiftAssignmentService do
+
+  describe "#perform" do
+    before do
+      @deliverers = FactoryGirl.create_list(:deliverer,3)
+      @shift = FactoryGirl.create(:shift)
+
+      @service1 = ShiftAssignmentService.new(@deliverers[0].id, @shift.id)
```

Comment:
instead of creating assignment through the service, we can directly use FactoryGirl
```ruby
FactoryGirl.create(:assignment, deliverer_id: @deliverers[0].id, shift_id: @shift.id)
```

Reply:
The intent of this spec is to test the output of `ShiftAssignmentService#perform` and `ShiftAssignmentService#errors` during the creation of `Assignment`.

Creating `Assignment` through `FactoryGirl` would bypass `#perform` and `#errors`. Should I leave it as it is or is there a better way to do it?

Reply:
Oh, sorry. I was looking at the invalid scenario.
For the success case, you're right, we need to use the service.
For the failure cases, we could use factory girl.